pkg/httpproxy/sign.go

```diff
@@ -101,9 +101,41 @@ func (a awsv4) sign(req *http.Request, secrets v1.SecretInterface, auth string)
 }
 
 func (a awsv4) getServiceAndRegion(host string) (string, string) {
-	//format : service.region.*
-	parts := strings.Split(host, ".")
-	return parts[0], parts[1]
+	service := ""
+	region := ""
+	for _, partition := range endpoints.DefaultPartitions() {
+		service, region = partitionServiceAndRegion(partition, host)
+		// empty region is valid, but if one is found it should be assumed correct
+		if region != "" {
+			return service, region
+		}
+	}
+
+	// if no region is found, global endpoint is assumed. In this case us-east-1 should be used for signing:
+	// https://docs.aws.amazon.com/general/latest/gr/sigv4_elements.html
+	return service, "us-east-1"
+}
+
+func partitionServiceAndRegion(partition endpoints.Partition, host string) (string, string) {
+	service := ""
+	for _, part := range strings.Split(host, ".") {
+		if id := partition.Services()[part].ID(); id != "" {
```

Comment:
* If host string is incorrectly set, and `part` doesn't exist in the map string, then the `.ID()` call here would fail. Should be able to solve with a second check. 
* Would be better to set `Services()` method above so you don't have to call it every loop.